File: app/scripts/ip_checker.rb
require './primer'

class IPChecker

  @monitor_file = './monitors/current_ip'

  def self.run
    require 'net/http'

    active_ip = Net::HTTP.get(URI.parse('http://ipecho.net/plain'))
    self.log_message('Active IP is ' + active_ip)

    if File.exists?(@monitor_file)
        recorded_ip = File.read(@monitor_file)
        self.log_message('Recorded IP is ' + recorded_ip.gsub("\n",''))
    else
        recorded_ip = nil
        self.log_message('No recorded IP.')
    end

    if active_ip.eql? recorded_ip
        self.log_message('No IP change')
    else
        self.log_message('IP address has changed')
    end

    fh = File.open(@monitor_file, 'w')
    fh.puts active_ip
    fh.close
  end

  def self.log_message message
    Primer.log_message 'ip_monitor_ruby', message
    puts message
  end

end

IPChecker.run

File: app/scripts/ip_checker.py
# Python 2.x
import urllib2
active_ip = urllib2.urlopen("http://ipecho.net/plain").read()

# Python 3.x
# import urllib.request
# urllib.request.urlopen("http://ipecho.com/plain").read()

print active_ip

try:
    fh = open('current_ip', 'r')
    recorded_ip = fh.readline()
    fh.close()
    print 'Recorded IP is ' + recorded_ip
except:
    print 'No recorded IP'
    recorded_ip = None
    pass

if active_ip == recorded_ip:
    print 'No IP change'
else:
    print 'New IP address is ' + active_ip

fh = open('current_ip', 'w')
fh.write(active_ip)
fh.close()
